iree/samples/static_library/CMakeLists.txt

```diff
@@ -10,8 +10,7 @@ if(NOT ${IREE_TARGET_BACKEND_DYLIB_LLVM_AOT}
   return()
```

Comment:
Line 9 should be updated to 

OR NOT (${IREE_BUILD_COMPILER} OR  DEFINED IREE_HOST_BINARY_ROOT)

If we want to support out-of-host compilation